mm2src/mm2_main/src/lp_swap/maker_swap_v2.rs

```diff
@@ -0,0 +1,898 @@
+use crate::mm2::lp_network::subscribe_to_topic;
+use crate::mm2::lp_swap::swap_v2_pb::*;
+use crate::mm2::lp_swap::{broadcast_swap_v2_msg_every, check_balance_for_maker_swap, recv_swap_v2_msg, SecretHashAlgo,
+                          SwapConfirmationsSettings, SwapsContext, TransactionIdentifier};
+use async_trait::async_trait;
+use bitcrypto::{dhash160, sha256};
+use coins::{ConfirmPaymentInput, FeeApproxStage, GenTakerPaymentSpendArgs, MarketCoinOps, MmCoin, SendPaymentArgs,
+            SwapOpsV2, TxPreimageWithSig};
+use common::log::{debug, info, warn};
+use common::{bits256, Future01CompatExt, DEX_FEE_ADDR_RAW_PUBKEY};
+use keys::KeyPair;
+use mm2_core::mm_ctx::MmArc;
+use mm2_number::MmNumber;
+use mm2_state_machine::prelude::*;
+use mm2_state_machine::storable_state_machine::*;
+use primitives::hash::H256;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use uuid::Uuid;
+
+// This is needed to have Debug on messages
+#[allow(unused_imports)] use prost::Message;
+
+/// If a certain P2P message is not received, swap will be aborted after this time expires.
+const NEGOTIATION_TIMEOUT_SEC: u64 = 90;
+
+/// Represents events produced by maker swap states.
+#[derive(Debug, PartialEq)]
+pub enum MakerSwapEvent {
+    /// Swap has been successfully initialized.
+    Initialized {
+        maker_coin_start_block: u64,
+        taker_coin_start_block: u64,
```

Comment:
These fields values are not known when a new state machine is created. We will have to initialize them with default values, or use `Option<u64>`. So, I preferred to duplicate the data instead of handling possible `None` or using `unwrap` like it's done in the current swap implementation:
https://github.com/KomodoPlatform/komodo-defi-framework/blob/d055cbaf6922adf828dc3f8058f2e7d4fe1a840d/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L1397